REGRESSION (Ventura): [ macOS ] 2X TestWebKitAPI.SOAuthorizationRedirect.DismissUIDuringMiniaturization (API-Tests) are constant failures

kkinnunen-apple opened the pull request:
#### da44a798c8a12288aacb2163909b6e3e859a4401
<pre>
REGRESSION (Ventura): [ macOS ] 2X TestWebKitAPI.SOAuthorizationRedirect.DismissUIDuringMiniaturization (API-Tests) are constant failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=247924">https://bugs.webkit.org/show_bug.cgi?id=247924</a>
rdar://problem/102343029

Reviewed by Darin Adler.

The test assumed that actions to NSWindows or NSApp reflect synchronously
back to the notifications. This is not the case.

Ensure that the test expects asynchronous operation.

Fix incorrect adoptNS() for notification center observers. The add function is
not a NS creation function, and as such it doesn&apos;t return an adoptable ref.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/256727@main">https://commits.webkit.org/256727@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac1937fb7f7e263eadae3e2d3e600b60a3da186f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5935 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106200 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/166515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100660 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6139 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34670 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89051 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102913 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102352 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/4585 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83280 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31548 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/86427 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88298 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74467 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40398 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/38059 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/21201 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4670 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/4329 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1144 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40483 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -129,6 +129,46 @@
 "</html>";
 
+static constexpr Seconds actionAbsenceTimeout = 300_ms;
+static constexpr Seconds actionDoneTimeout = 1000_ms;
```
No practical reason to write `static` as well as `constexpr`.

kkinnunen-apple replied:
Thanks, removed inline. I now know that constexpr has internal linkage unless `inline` is specified.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -129,6 +129,46 @@
+            return;
+        [m_center removeObserver:m_observer.get()];
+    }
+    ScopedNotificationCenterObserveOnce(ScopedNotificationCenterObserveOnce&&) = default;
+    ScopedNotificationCenterObserveOnce& operator=(ScopedNotificationCenterObserveOnce&&) = default;
```
Why did you have to write these? Won’t it compile and do exactly the same thing without explicitly defining them?

kkinnunen-apple replied:
I was actually trying to prevent copying but allow moving, but forgot the `WTF_NONCOPYABLE`. Thanks, added.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -129,6 +129,46 @@
+        return TestWebKitAPI::Util::runFor(&m_didObserve, timeout);
+    }
+
+    operator bool() const { return m_didObserve; }
```
I think you should write `explicit operator bool`.

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -129,6 +129,46 @@
+
+    operator bool() const { return m_didObserve; }
+private:
+    NSNotificationCenter * __weak m_center = [NSNotificationCenter defaultCenter];
```
We’ve been using `{ }` rather than `=` for this kind of initialization. (If it was my own personal project I might have chosen differently.)

darinadler reviewed (approved):
Looks great

kkinnunen-apple committed:
REGRESSION (Ventura): [ macOS ] 2X TestWebKitAPI.SOAuthorizationRedirect.DismissUIDuringMiniaturization (API-Tests) are constant failures
https://bugs.webkit.org/show_bug.cgi?id=247924
rdar://problem/102343029

Reviewed by Darin Adler.

The test assumed that actions to NSWindows or NSApp reflect synchronously
back to the notifications. This is not the case.

Ensure that the test expects asynchronous operation.

Fix incorrect adoptNS() for notification center observers. The add function is
not a NS creation function, and as such it doesn't return an adoptable ref.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/256727@main
